server/src/main/java/org/elasticsearch/Version.java

```diff
@@ -163,6 +163,11 @@
                 + org.apache.lucene.util.Version.LATEST + "] is still set to [" + CURRENT.luceneVersion + "]";
     }
 
+    private static final List<Version> DECLARED_VERSIONS;
+    static {
+        DECLARED_VERSIONS = getDeclaredVersions(Version.class);
```

Comment:
Please make this lazy, and make this an unmodifiable list. Also, I think we can drop this method and only access the list?

Reply:
I think the method needs to stay since it can be called with a different version class:
https://github.com/elastic/elasticsearch/blob/master/test/framework/src/main/java/org/elasticsearch/test/VersionUtils.java#L53

Reply:
Yet you can expose the list to that class too?

Reply:
The input in the VersionUtils method is an arbitrary `Class<?>` rather than `Version.class`, so the values in the list will change based on the input.

Reply:
I see, thanks for investigating.